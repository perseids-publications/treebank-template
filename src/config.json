{
  "title": "Treebank Publication Template",
  "subtitle": "Template for publishing collections of treebanks. See the [repository](https://github.com/perseids-publications/treebank-template/) for more information and documentation. If you would like to create a template with your own treebanks, send us a [message](https://docs.google.com/forms/d/e/1FAIpQLSf8yf6B3xlV31x0JKquRuIdXd6LVuX0V7I7id7ZLXbMwDo-UA/viewform).",
  "doi": "10.5281/zenodo.3827930",
  "copyright": "The Perseids Project",
  "report": "https://github.com/perseids-publications/treebank-template/issues",
  "github": "https://github.com/perseids-publications/treebank-template/",
  "twitter": "https://twitter.com/PerseidsProject",
  "collections": [
    {
      "title": "Information",
      "text": "The Treebank Template is maintained by the Perseids Project. See the [getting started instructions](/instructions/getting-started/). Or visit [our homepage](https://www.perseids.org) to learn more."
    },
    {
      "title": "Collection 1",
      "publications": [
        {
          "path": "on-the-murder-of-eratosthenes",
          "author": "Lysias",
          "work": "On the Murder of Eratosthenes",
          "editors": "Vanessa Gorman",
          "sections": [
            {
              "locus": "1-50",
              "path": "on-the-murder-of-eratosthenes-1-50",
              "xml": "lysias-1-1-50.xml",
              "link": "https://www.example.com",
              "notes": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
              "chunks": { "start": 1, "end": 134 }
            }
          ]
        },
        {
          "path": "on-the-crown",
          "author": "Demosthenes",
          "work": "On the Crown",
          "editors": "Vanessa Gorman",
          "sections": [
            {
              "locus": "1-50",
              "path": "on-the-crown-1-50",
              "xml": "demosthenes-18-1-50.xml",
              "license": "CC BY-SA 4.0",
              "notes": "Lorem ipsum dolor sit amet, *consectetur* adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
              "chunks": { "start": 1, "end": 160 }
            },
            {
              "locus": "51-100",
              "path": "on-the-crown-51-100",
              "xml": "demosthenes-18-51-100.xml",
              "chunks": { "start": 1, "end": 149 }
            }
          ]
        }
      ]
    },
    {
      "title": "Collection 2",
      "publications": [
        {
          "path": "histories",
          "author": "Herodotus",
          "work": "The Histories",
          "editors": "Vanessa Gorman",
          "sections": [
            {
              "locus": "40-59",
              "path": "histories-40-59",
              "xml": "herodotus-1-40-59.xml",
              "chunks": { "start": 1, "end": 100 }
            },
            {
              "locus": "60-79",
              "path": "histories-60-79",
              "xml": "herodotus-1-60-79.xml",
              "chunks": { "start": 1, "end": 166 }
            },
            {
              "locus": "80-99",
              "path": "histories-80-99",
              "xml": "herodotus-1-80-99.xml",
              "notes": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
              "chunks": { "start": 1, "end": 185 }
            }
          ]
        },
        {
          "path": "philippic",
          "author": "Demosthenes",
          "work": "Philippic 1",
          "editors": [
            "Robert Gorman",
            "Vanessa Gorman"
          ],
          "sections": [
            {
              "locus": "1-51",
              "path": "philippic-1-51",
              "xml": "demosthenes-4-phil1-bu1.xml",
              "chunks": { "numbers": ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "41", "42", "43", "44", "45", "46", "47", "48", "49", "50", "51", "52", "53", "54", "55", "56", "57", "58", "59", "60", "61", "62", "63", "64", "65", "66", "67", "68", "69", "70", "71", "72", "73", "74", "75", "76", "77", "78", "79", "80", "81", "82", "83", "84", "85", "86", "87", "88", "89", "90", "91", "92", "93", "94", "95", "96", "97", "98", "99", "100", "101", "102", "103", "104", "105", "106", "107", "108", "109", "110", "111", "112", "113", "114", "115", "116", "117", "118", "119", "120", "121", "122", "123", "124", "125", "126", "127", "128", "129", "130", "131", "132", "133", "134", "135", "136", "137", "138", "139", "140", "141", "142", "143", "144", "145", "146", "147", "148", "149", "150", "151", "152", "153", "154", "155", "156", "157", "158", "159"] }
            }
          ]
        }
      ]
    }
  ]
}
