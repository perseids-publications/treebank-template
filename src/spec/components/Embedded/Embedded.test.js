import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { render, waitForElementToBeRemoved } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import config from '../../config.test.json';
import treebanks from '../../treebanks.test.json';

import Embedded from '../../../components/Embedded';
// import TreebankService from '../../../components/TreebankService';

jest.mock('treebank-react');

const server = setupServer(
  rest.get(`${process.env.PUBLIC_URL}/xml/lysias-1-1-50.xml`, (req, res, ctx) => (
    res(ctx.delay(10), ctx.text(treebanks.lysias))
  )),
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

it('renders an embedded publication', async () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container, queryByText } = render(component);

  expect(container).toMatchSnapshot();

  await waitForElementToBeRemoved(() => queryByText('Loading...'));

  expect(container).toMatchSnapshot();
});

// it('provides an API to communicate with an embedded publication', () => {
//   const component = (
//     <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
//       <Embedded config={config} />
//     </MemoryRouter>
//   );
//   const tree = renderer.create(component);
//   const { instance: { messageHandler } } = tree.root.findByType(TreebankService);
//
//   messageHandler(
//     { ID: 'test', body: { gotoSentence: { sentenceId: '5' } } },
//     () => {},
//   );
//
//   expect(tree).toMatchSnapshot();
// });
//
// it('the API can be used to select words', () => {
//   const component = (
//     <MemoryRouter initialEntries={['/embed/on-the-murder-of-eratosthenes-1-50/1']}>
//       <Embedded config={config} />
//     </MemoryRouter>
//   );
//   const tree = renderer.create(component);
//   const { instance: { messageHandler } } = tree.root.findByType(TreebankService);
//
//   messageHandler(
//     { ID: 'test', body: { gotoSentence: { sentenceId: '5', wordIds: ['12'] } } },
//     () => {},
//   );
//
//   expect(tree).toMatchSnapshot();
// });

it('renders 404 when publication not found', () => {
  const component = (
    <MemoryRouter initialEntries={['/embed/unknown']}>
      <Embedded config={config} />
    </MemoryRouter>
  );
  const { container } = render(component);

  expect(container).toMatchSnapshot();
});
