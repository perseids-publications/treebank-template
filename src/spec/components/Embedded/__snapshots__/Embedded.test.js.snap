// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 404 when publication not found 1`] = `
<div>
  <div
    class="container pt-5"
  >
    <div
      class="row col-12 pt-5 pb-3"
    >
      <div
        class="col-12 text-center"
      >
        <h1>
          Error 404
        </h1>
      </div>
    </div>
    <div
      class="row col-12 pb-3"
    >
      <div
        class="col-12 text-center"
      >
        <h2>
          Publication not found
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders an embedded alignment publication 1`] = `
<div>
  <div>
    <div
      class="alignmentContainer"
    >
      <div
        class="text"
      >
        <div
          class="segment"
        >
          <div
            class="_2vy3L"
          >
            <p>
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                Ἡροδότου
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                Ἁλικαρνησσέος
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἱστορίης
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἀπόδεξις
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἥδε
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ὡς
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                μήτε
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τὰ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                γενόμενα
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἐξ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἀνθρώπων
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τῷ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                χρόνῳ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἐξίτηλα
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                γένηται
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                μήτε
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἔργα
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                μεγάλα
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τε
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                καὶ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                θωμαστά
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τὰ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                μὲν
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                Ἕλλησι
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τὰ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                δὲ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                βαρβάροισι
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἀποδεχθέντα
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἀκλεᾶ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                γένηται
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τά
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                τε
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἄλλα
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                καὶ
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                δι᾽
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἣν
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                αἰτίην
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἐπολέμησαν
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ἀλλήλοισι
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                .
              </span>
            </p>
          </div>
        </div>
        <div
          class="segment"
        >
          <div
            class="_2vy3L"
          >
            <p>
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                This
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                is
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                the
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                display
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                of
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                the
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                inquiry
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                of
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                Herodotus
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                of
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                Halicarnassus
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                so
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                that
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                things
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                done
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                by
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                man
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                not
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                be
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                forgotten
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                in
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                time
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                and
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                that
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                great
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                and
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                marvelous
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                deeds
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                some
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                displayed
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                by
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                the
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                Hellenes
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                some
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                by
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                the
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                barbarians
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                not
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                lose
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                their
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                glory
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                ,
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                including
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                among
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                others
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                what
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                was
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                the
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                cause
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                of
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                their
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                waging
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                war
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                on
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                each
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                other
              </span>
               
              <span
                class="_2xbMK"
                role="button"
                tabindex="0"
              >
                .
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders an embedded publication 1`] = `
<div>
  <div>
    <div>
      Loading...
    </div>
  </div>
</div>
`;

exports[`renders an embedded publication 2`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight=""
        data-id="1"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the API performs a no-op for refreshView 1`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight=""
          data-id="1"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight=""
        data-id="1"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the API supports gotoSentence 1`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight=""
          data-id="5"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight=""
          data-id="5"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight=""
        data-id="5"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/5"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`the API supports gotoSentence with highlighted words 1`] = `
<div>
  <div>
    <div
      class="treebankContainer"
    >
      <div
        class="text"
      >
        <div
          data-highlight="12"
          data-id="5"
          data-treebank="1912971948"
        >
          Text
        </div>
      </div>
      <div
        class="graph"
      >
        <div
          data-highlight="12"
          data-id="5"
          data-treebank="1912971948"
        >
          Graph
        </div>
      </div>
      <div
        data-highlight="12"
        data-id="5"
        data-treebank="1912971948"
      >
        PartOfSpeech
      </div>
      <div
        class="links"
      >
        <p>
          <a
            href="https://www.example.com/on-the-murder-of-eratosthenes-1-50/5"
            rel="noopener noreferrer"
            target="_blank"
          >
            Credits and more information
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
